datafusion/core/src/physical_plan/insert.rs

```diff
@@ -219,3 +237,25 @@ fn make_count_schema() -> SchemaRef {
         false,
     )]))
 }
+
+fn check_batch(batch: RecordBatch, schema: &SchemaRef) -> Result<RecordBatch> {
```

Comment:
Im not really sure about this method. That might be a performance hit if we scan data.
I'd rather check metadata of incoming schema and expected schema like Spark does.

Reply:
@comphead Thank you. There are some scenarios where calling this function is not necessary. I have improved it.